### PR TITLE
ipv6_addr: make IPv4 compatible addresses optional

### PR DESCRIPTION
**Pull request by authmillenon**
Asked by @haukepetersen and @cgundogan (and others) in #3608, now delivered

~~Depends on #3608.~~

### CODE REVIEW

**Comment by cgundogan** [comment]
rebase to current master please

**Comment by authmillenon** [comment]
Rebased to current master

**Comment by authmillenon** [comment]
Test with

```bash
make -C tests/unittests tests-ipv6_addr test
USEMODULE=ipv4_addr make -C tests/unittests clean tests-ipv6_addr test
```

File: sys/include/net/ipv6/addr.h [hunk 1]
```diff
@@ -40,11 +40,18 @@ extern "C" {
...
+#ifdef MODULE_IPV4_ADDR
 /**
  * @brief   Maximum length of an IPv6 address as string.
  */
-#define IPV6_ADDR_MAX_STR_LEN       (sizeof("ffff:ffff:ffff:ffff:" \
-                                            "ffff:ffff:255.255.255.255"))
+#define IPV6_ADDR_MAX_STR_LEN       (sizeof("ffff:ffff:ffff:ffff:ffff:ffff:255.255.255.255"))
+#else
+/**
+ * @brief   Maximum length of an IPv6 address as string.
```
Comment (cgundogan) [inline sys/include/net/ipv6/addr.h]: the second `@brief` section is not necessary IMO

**Comment by cgundogan** [comment]
I tested with the existing unittest and with `gnrc_example`. Building without `ipv4_addr` cuts 400 bytes of ROM for the samr21-xpr in the `gnrc_example` :+1:

File: sys/include/net/ipv6/addr.h (hunk 1 above)
Comment (authmillenon) [inline sys/include/net/ipv6/addr.h]: You did not even give me time to fix this :(

**Comment by authmillenon** [comment]
> I tested with the existing unittest and with gnrc_example. Building without ipv4_addr cuts 400 bytes of ROM for the samr21-xpro in the gnrc_example :+1:

Building with `ENABLE_DEBUG` should also be a little smaller now, since the `addr_str` buffers are a little shorter without `ipv4_addr` now.

**Pull request merged**